### PR TITLE
Use pooled memory for the streamwriter

### PR DESCRIPTION
**Pull request by rynowak**
/cc @pranavkm @kichalla

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Core/HttpResponseStreamWriter.cs [hunk 1]
```diff
@@ -168,6 +216,16 @@ public override Task FlushAsync()
         protected override void Dispose(bool disposing)
         {
             FlushInternal(flushStream: false, flushEncoder: true);
+
+            if (_leasedByteBuffer != null)
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Core/HttpResponseStreamWriter.cs]: `_leasedByteBuffer?.Owner.Return(_leasedByteBuffer);`

**Comment by pranavkm** [comment]
:shipit:

File: src/Microsoft.AspNet.Mvc.Core/HttpResponseStreamWriter.cs (hunk 1 above)
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/HttpResponseStreamWriter.cs]: Wouldn't this result in a nullref?
```
var temp = _leasedByteBuffer?.Owner;
temp.Return(_leasedByteBuffer);
```

**Comment by rynowak** [comment]
Some perf numbers for this:

**Before**
image


**After**
image

Allocations for 3001 requests to a site with model binding and a very minimal view.

**Pull request closed without merging**